src/content-scripts/index.js

```diff
@@ -84,12 +84,28 @@ class EventRecorder {
     }
   }
 
+  getDataAttributeContainer (element) {
+    if (!element) return
+
+    if (element.attributes[this.dataAttribute]) {
+      return element
+    }
+
+    if (element.parentElement) {
+      return this.getDataAttributeContainer(element.parentElement)
```

Comment:
Should we place any rule on this recursion? A magic number of maximum depth wouldn't make sense in my opinion, and I couldn't think about any condition to end the recursion. Any ideas?